### PR TITLE
Missing Inverse Attributes and Spatial Interference CTs

### LINKED ISSUE #218: Spatial Interference Templates
## Description & Tasks
Documentation of structure to represent spatial interference with the project context including sub templates relating to the use of Spatial Zone.

- [ ] Author Spatial Interference Template
- [ ] Author Spatial Interfernce with Zone Template
- [ ] Detailed Documentation per Solution on MIRO

### Discussion Points

1. We need 2 templates for using 1 for IfcRelInterferesElement and a subtemplate for IfcSpatailZone - Agree/Disagree?
2. Do we need additional templates for the footprint geometry solutions or will GU suffice?
3. AOB?

### Links
[Miro Element](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362978830650&cot=10)
[Miro Task Card](https://miro.com/app/board/o9J_l5rJCLk=/?moveToWidget=3074457362911087561&cot=14)
[Spatial Structure Miro](https://miro.com/app/board/o9J_l94erOI=/)

### Decisions 
TBC

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
Added mirroring inverse attributes missing on IfcSpatialElement from IfcElement.

Fixes #218 
First draft of Spatial Interference and Spatial Interference with Zones Concept templates

### CODE REVIEW

**Review by MatthieuPERIN — APPROVED** [review]
Perfect for me !

File: IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcSpatialElement/DocEntity.xml [hunk 1]
```diff
@@ -37,6 +37,20 @@
...
+&gt; NOTE&amp;nbsp; There is no indication of precedence between _IsInterferedByElements_ and _InterferesElements_ orientated interference is defined by _IfcRelInterferesElements.ImpliedOrder_ or _IfcRelInterferesElements.InterferenceType_.
+
+{ .change-ifc2x4}
+&gt; IFC4x3 CHANGE New inverse relationship.</Documentation>
+		</DocAttribute>
+		<DocAttribute Name="InterferesElements" UniqueId="d0a42a83-3d1e-4f71-b10f-d1b1e14fc808" DefinedType="IfcRelInterferesElements" AggregationType="3" Inverse="RelatingElement">
+			<Documentation>Reference to the interference relationship to indicate the spatial element that interferes. The relationship, if provided, indicates that this spatial element has an interference with one or many other spatial elements.
+&gt; NOTE&amp;nbsp; There is no indication of precedence between _IsInterferedByElements_ and _InterferesElements_ orientated interference is defined by _IfcRelInterferesElements.ImpliedOrder_ or _IfcRelInterferesElements.InterferenceType_.
```
Comment (larswik) [inline IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcSpatialElement/DocEntity.xml:49]:
```suggestion
&gt; NOTE&amp;nbsp; There is no indication of precedence between _IsInterferedByElements_ and _InterferesElements_. Orientated interference is defined by _IfcRelInterferesElements.ImpliedOrder_ or _IfcRelInterferesElements.InterferenceType_.
```

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Documentation.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+The _Spatial Interference_ concept defines the relationship of spatial elements, such as facilities (_IfcFacility_ & specialised subtypes) or faclity parts (_IfcFacilityPart_ with domain specific predefined types) that interfere or interface with other spatial elements across discipline spatial hierarchy branches. The _IfcSpatialElement_s should be identifiable by their _IfcSpatialElement.Name_ attribute and an optional _IfcRelInterferesElements.InterferenceType_ should be included to describe the nature of the interface or interference.
```
Comment (larswik) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Documentation.md:1]:
```suggestion
The _Spatial Interference_ concept defines the relationship of spatial elements, such as facilities (_IfcFacility_ & specialised subtypes) or facility parts (_IfcFacilityPart_ with domain specific predefined types) that interfere or interface with other spatial elements across discipline spatial hierarchy branches. The _IfcSpatialElement_s should be identifiable by their _IfcSpatialElement.Name_ attribute and an optional _IfcRelInterferesElements.InterferenceType_ should be included to describe the nature of the interface or interference.
```

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Documentation.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+
+A simple example is a project that contains a road or railway development that includes a bridge section. _IfcRelInterferesElements_ is used to semantically link the spatial segments of the road or railway that pass over the bridge or bridge segments with the relevent interference type specified. This semantic relationship provides an easily queriable connection to identify the spatial elements that require consideration across disciplines such as the road design team and bridge design team. How the spatial hierarchy is organised is up to the user and project in question.
```
Comment (larswik) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Documentation.md:3]:
```suggestion
A simple example is a project that contains a road or railway development that includes a bridge section. _IfcRelInterferesElements_ is used to semantically link the spatial segments of the road or railway that pass over the bridge or bridge segments with the relevant interference type specified. This semantic relationship provides an easily queriable connection to identify the spatial elements that require consideration across disciplines such as the road design team and bridge design team. How the spatial hierarchy is organised is up to the user and project in question.
```

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+The _Spatial Interference with Zone_ concept is an extension of the _Spatial Inteference_ concept which defines the relationship of spatial elements, such as facilities (_IfcFacility_ & specialised subtypes) or faclity parts (_IfcFacilityPart_ with domain specific predefined types) that interfere or interface with other spatial elements across discipline spatial hierarchy branches, and extends the dataset with an related inteference _IfcSpatialZone_ via the _IfcRelInterferesElements.InterferenceSpace_ attribute. The realizing _IfcSpatialZone_ should always have the _PredefinedType_ value set to INTERFERENCE.
+
+The addition of a realizing _ifcSpatialZone_ extends the functionality of this relationship with the following:
+- The ability to attach property sets to the co-engineering (or interference) zone.
+- Allows overlapping footprint or body geometry to remain relevent to the _IfcSpatialElement_s and then define an explicit shared  footprint or body geometry.
```
Comment (larswik) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md:6]:
```suggestion
- Allows overlapping footprint or body geometry to remain relevant to the _IfcSpatialElement_s and then define an explicit shared  footprint or body geometry.
```

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/DocTemplateDefinition.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+										<DocModelRuleAttribute Name="PredefinedType" Identification="SpatialZoneType">
+											<Rules>
+												<DocModelRuleEntity id="IfcSpatialZoneTypeEnum_0000000000000000000000" Name="IfcSpatialZoneTypeEnum" Description="*">
+													<Rules>
+														<DocModelRuleConstraint>
+															<Expression xsi:type="DocOpStatement" Operation="compareequal">
+																<Reference xsi:type="DocOpReference">
+																	<EntityRule xsi:type="DocModelRuleEntity" xsi:nil="true" href="IfcSpatialZoneTypeEnum_0000000000000000000000" />
+																</Reference>
+																<Value xsi:type="DocOpLiteral" Operation="loadstring" Literal="INTERFERENCE" />
```
Comment (larswik) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/DocTemplateDefinition.xml:39]: Just a check out of ignorance that this is the correct syntax, and that it should not be ".INTERFERENCE." as in the STEP format?

**Review by larswik — CHANGES_REQUESTED** [review]

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md (hunk 4 above)
Comment (larswik) [reply IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md:6]: Sentence feels a little bit strange to me. Could it be "Allows explicit definition of the shared footprint or body geometry representing the zone where the _IfcSpatialElement_s interfere" or something similar?

Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md]:
```suggestion
- Allows explicit definition of the shared footprint or body geometry representing the zone where the _IfcSpatialElement_s interfere, without impacting the footprint or body geometry of the interfering _IfcSpatialElement_s
```

Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md]:
```suggestion
- Allows explicit definition of the shared footprint or body geometry representing the zone where the _IfcSpatialElement_s interfere, without impacting the footprint or body geometry of the interfering _IfcSpatialElement_s
```

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/DocTemplateDefinition.xml (hunk 5 above)
Comment (AlexBrad1eyCT) [reply IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/DocTemplateDefinition.xml:39]: @SergejMuhic, can you double check my syntax on this.

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+
+The _Spatial Interference with Zone_ concept is an extension of the _Spatial Inteference_ concept which defines the relationship of spatial elements, such as facilities (_IfcFacility_ & specialised subtypes) or faclity parts (_IfcFacilityPart_ with domain specific predefined types) that interfere or interface with other spatial elements across discipline spatial hierarchy branches, and extends the dataset with an related inteference _IfcSpatialZone_ via the _IfcRelInterferesElements.InterferenceSpace_ attribute. The realizing _IfcSpatialZone_ should always have the _PredefinedType_ value set to INTERFERENCE.
+
+The addition of a realizing _ifcSpatialZone_ extends the functionality of this relationship with the following:
+- The ability to attach property sets to the co-engineering (or interference) zone.
+- Allows overlapping footprint or body geometry to remain relevant to the _IfcSpatialElement_s and then define an explicit shared  footprint or body geometry.
```
Comment (AlexBrad1eyCT) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md:6]:
```suggestion
- Allows explicit definition of the shared footprint or body geometry representing the zone where the _IfcSpatialElement_s interfere, without impacting the footprint or body geometry of the interfering _IfcSpatialElement_s
```

**Comment by AlexBrad1eyCT** [comment]
@MatthieuPERIN @larswik, Made the spelling & Grammar changes, and adjusted one sentance

**Review by larswik — APPROVED** [review]

**Review by MatthieuPERIN — APPROVED** [review]

**Review by jmirtsch — CHANGES_REQUESTED** [review]
Fix UniqueIds

**Review by MatthieuPERIN — APPROVED** [review]

**Review by larswik — APPROVED** [review]

**Pull request merged**